### ISSUE TITLE
[influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'

### ISSUE THREAD

**Issue by photis** [bernd/statsd-influxdb-backend#30]
Linux 4.1.15-v7+ #830 SMP Tue Dec 15 17:02:45 GMT 2015 armv7l GNU/Linux
Node v4.2.4
NPM 2.14.12
Influxdb v0.9.6
Statsd 0.7.2 (even though NPM reports statsd@0.7.1)

My config.js file:

```
{
    port: 8125,
    backends: [ "statsd-influxdb-backend" ],

    influxdb: {
        host: '127.0.0.1',
        port: 8086,
        database: 'statsd',
        username: 'user',
        password: 'pass',
        version: 0.9,
        flush: {
            enable: true
        },
        proxy: {
            enable: false,
            suffix: 'raw',
            flushInterval: 1000
        },
        includeStatsdMetrics: true,
        includeInfluxdbMetrics: true,
    },

    debug: true,
    legacyNamespace: false
}
```

When I start statsd I get:

```
10 Jan 23:49:48 - reading config file: /opt/statsd/config.js
10 Jan 23:49:48 - server is up
10 Jan 23:49:48 - DEBUG: Loading backend: statsd-influxdb-backend
10 Jan 23:49:58 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
10 Jan 23:50:08 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
10 Jan 23:50:18 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
10 Jan 23:50:28 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
...and so on, forever.
```

No data flows from statsd to influxdb.
And, yes, the following command works:

```
curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE mydb"
```

I installed the backend as follows:

```
$ sudo npm install statsd-influxdb-backend -d
npm info it worked if it ends with ok
npm info using npm@2.14.12
npm info using node@v4.2.4
npm WARN package.json statsd@0.7.1 No license field. 
npm info addNameTag [ 'statsd-influxdb-backend', 'latest' ]
npm info attempt registry request try #1 at 3:20:23 PM
npm http request GET https://registry.npmjs.org/statsd-influxdb-backend
npm http 200 https://registry.npmjs.org/statsd-influxdb-backend
npm info retry fetch attempt 1 at 3:20:24 PM
npm info attempt registry request try #1 at 3:20:24 PM
npm http fetch GET https://registry.npmjs.org/statsd-influxdb-backend/-/statsd-influxdb-backend-0.6.0.tgz
npm http fetch 200 https://registry.npmjs.org/statsd-influxdb-backend/-/statsd-influxdb-backend-0.6.0.tgz
npm info install statsd-influxdb-backend@0.6.0 into /opt/statsd
npm info installOne statsd-influxdb-backend@0.6.0
npm info preinstall statsd-influxdb-backend@0.6.0
npm info build /opt/statsd/node_modules/statsd-influxdb-backend
npm info linkStuff statsd-influxdb-backend@0.6.0
npm info install statsd-influxdb-backend@0.6.0
npm info postinstall statsd-influxdb-backend@0.6.0
statsd-influxdb-backend@0.6.0 node_modules/statsd-influxdb-backend
npm info ok 
$
```

I'm pretty new to all this. Does anyone know a solution for it?

**Comment by kapilratnani** [comment]
According to quick googling i found that this issue comes if you are using
old version of nodejs.
Anyway can you share how you are starting statsd? May be we can find
something from that.

On Mon, Jan 11, 2016 at 2:59 PM, oriste notifications@github.com wrote:

> Linux 4115-v7+ #830 SMP Tue Dec 15 17:02:45 GMT 2015 armv7l GNU/Linux
> Node v424
> NPM 21412
> Influxdb v096
> Statsd 072 (even though NPM reports statsd@071)
> 
> My configjs file:
> 
> {
>     port: 8125,
>     backends: [ "statsd-influxdb-backend" ],
> 
> ```
> influxdb: {
>     host: '127001',
>     port: 8086,
>     database: 'statsd',
>     username: 'user',
>     password: 'pass',
>     version: 09,
>     flush: {
>         enable: true
>     },
>     proxy: {
>         enable: false,
>         suffix: 'raw',
>         flushInterval: 1000
>     },
>     includeStatsdMetrics: true,
>     includeInfluxdbMetrics: true,
> },
> 
> debug: true,
> legacyNamespace: false
> ```
> 
> }
> 
> When I start statsd I get:
> 
> 10 Jan 23:49:48 - reading config file: /opt/statsd/configjs
> 10 Jan 23:49:48 - server is up
> 10 Jan 23:49:48 - DEBUG: Loading backend: statsd-influxdb-backend
> 10 Jan 23:49:58 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
> 10 Jan 23:50:08 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
> 10 Jan 23:50:18 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
> 10 Jan 23:50:28 - [influxdb] TypeError: Object #<EventEmitter> has no method 'hrtime'
> and so on, forever
> 
> No data flows from statsd to influxdb
> And, yes, the following command works:
> 
> curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE mydb"
> 
> I installed the backend as follows:
> 
> $ sudo npm install statsd-influxdb-backend -d
> npm info it worked if it ends with ok
> npm info using npm@21412
> npm info using node@v424
> npm WARN packagejson statsd@071 No license field
> npm info addNameTag [ 'statsd-influxdb-backend', 'latest' ]
> npm info attempt registry request try #1 at 3:20:23 PM
> npm http request GET https://registrynpmjsorg/statsd-influxdb-backend
> npm http 200 https://registrynpmjsorg/statsd-influxdb-backend
> npm info retry fetch attempt 1 at 3:20:24 PM
> npm info attempt registry request try #1 at 3:20:24 PM
> npm http fetch GET https://registrynpmjsorg/statsd-influxdb-backend/-/statsd-influxdb-backend-060tgz
> npm http fetch 200 https://registrynpmjsorg/statsd-influxdb-backend/-/statsd-influxdb-backend-060tgz
> npm info install statsd-influxdb-backend@060 into /opt/statsd
> npm info installOne statsd-influxdb-backend@060
> npm info preinstall statsd-influxdb-backend@060
> npm info build /opt/statsd/node_modules/statsd-influxdb-backend
> npm info linkStuff statsd-influxdb-backend@060
> npm info install statsd-influxdb-backend@060
> npm info postinstall statsd-influxdb-backend@060
> statsd-influxdb-backend@060 node_modules/statsd-influxdb-backend
> npm info ok
> $
> 
> I'm pretty new to all this Does anyone know a solution for it?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/bernd/statsd-influxdb-backend/issues/30.

## 

Kapil Ratnani
MT2009119
M.Tech(IT) IIIT Bangalore
Senior Software Developer
Myntra Designs Pvt. Ltd.

**Comment by photis** [comment]
Yes, I googled the same, which left me even more confused as I use the latest version already (v4.2.4 for armv7l).

But you're an **ACE**!!!!

I used the example statsd.sh (supplied with the package) to create a daemon process (in /etc/init.d) and start/restart/stop if necessary with `sudo service statsd start/restart/stop` as appropriate.
You asking for this however made me try `node /opt/statsd/stats.js /opt/statsd/config.js` directly, just out of curiosity, and lo-and-behold, no more error message and data is flowing into my Grafana dashboard! 

It's so beautiful! Thank you, thank you, thank you!

I'll look into a better way to make this a service.
